source/system/native/MutexUnix.ooc

```diff
@@ -19,7 +19,17 @@ version(unix || apple) {
 			super()
 		}
 		lock: override func {
-			pthread_mutex_lock(this _backend&)
+			version(debugDeadlock) {
+				timeLimitMilliseconds := 1000 //Only whole seconds
```

Comment:
I don't understand what you mean. Isn't this what the `tv_sec` and `tv_nsec` variables are for? Something like
```ooc
deadline tv_sec = currentTime tv_sec + timeLimitMilliseconds / 1000
deadline tv_nsec = currentTime tv_usec * 1000 + (timeLimitMilliseconds % 1000) * 1_000_000
```